### PR TITLE
Milestone 4: Controllers spec

### PR DESCRIPTION
**Pull request by Christianib003**
- [x] Add RSpec as a gem to your project.
- [x] Create a Request spec file for all your controllers.
_Make sure that for each action you check:_
    - [x] If response status was correct.
    - [x] If a correct template was rendered.
    - [x] If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -3,8 +3,23 @@
...
+    it 'returns the http sucess' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'displays the intended template' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to render_template('show')
+    end
   end
```
Comment (gemmen29) [inline spec/requests/posts_spec.rb:34]: - [ ] Kindly include tests for the `Post#index` action that represents If the response body includes correct placeholder text as per this [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.).

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -3,8 +3,23 @@
...
+
+    it 'returns the http sucess' do
+      get '/users/:id'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'displays the intended template' do
+      get '/users/:id'
+      expect(response).to render_template('show')
+    end
```
Comment (gemmen29) [inline spec/requests/users_spec.rb:28]: - [ ] Kindly include tests for the `User#index` action that represents If the response body includes correct placeholder text as per this [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.).

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -3,8 +3,23 @@
...
+
+    it 'returns the http sucess' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'displays the intended template' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to render_template('show')
+    end
```
Comment (gemmen29) [inline spec/requests/posts_spec.rb:28]: - [ ] Kindly include tests for the `Post#show` action that represents If the response body includes correct placeholder text. as per this [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.).

File: spec/requests/users_spec.rb (hunk 2 above)
Comment (gemmen29) [inline spec/requests/users_spec.rb:28]: - [ ] Kindly include tests for the `User#show` action that represents If the response body includes correct placeholder text. as per this [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Christianib003, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/users_spec.rb (hunk 2 above)
Comment (Christianib003) [reply spec/requests/users_spec.rb:28]: Hello @gemmen29 ;

I appreciate your review on this projet. however the issues stated i think I had already implemented them. There are different `get` for each two tests (`get '/users/:id'` for `user#show` which uses `show.html.erb` as the template for verifying both placeholder and template itself, `get '/users'` for `user#index` which uses `index.html.erb as template`, also `get '/users/:user_id/posts/'` for `post#index` which uses a different `index.html.erb`  finally `get '/users/:user_id/posts/:id'` which uses a different `show.html.erb` as it acan be seen in the `routes.rb` file ). Check it out please. thanks!

Comment (gemmen29) [reply spec/requests/users_spec.rb:28]: Hi @Christianib003 You should implement 3 tests for each action in both user and post controller as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.). so the total tests for each one of them will be 6 tests per controller and 12 tests total.

Comment (Christianib003) [reply spec/requests/users_spec.rb:28]: @gemmen29 

The way I did Is that the test for template received both the template and the placeholder text in form of a file which either index or show and one test actually carries out two tasks. Which reduces the number of tests

Comment (gemmen29) [reply spec/requests/users_spec.rb:28]: @Christianib003 
I see you already did the status code test for the `User#index` and `Post#index` actions and I edited my previous comments but you still didn't do the placeholder text test for any action. The placeholder test is to check if the view includes a specific text.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Christianib003, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**